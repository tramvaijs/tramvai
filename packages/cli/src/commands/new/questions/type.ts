import any from '@tinkoff/utils/array/any';

type Type = 'app' | 'child-app';

const choices = [
  {
    name: 'Standalone tramvai app',
    value: 'app',
  },
  {
    name: 'Microfrontend implemented by tramvai child-app',
    value: 'child-app',
  },
];

const typeQuestion = (answer: string) => ({
  type: 'list' as const,
  name: 'type' as const,
  message: 'Choose a type for your project',
  choices,
  when: () => !any<{ name: string; value: string }>((choice) => choice.value === answer)(choices),
  default: 'app' as const,
});

export { Type, typeQuestion };
