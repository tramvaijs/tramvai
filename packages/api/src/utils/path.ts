import path from 'node:path';

export const resolveAbsolutePathForFile = ({
  file,
  sourceDir,
  rootDir,
}: {
  file: string;
  sourceDir: string;
  rootDir: string;
}): string => {
  if (path.isAbsolute(file)) {
    return file;
  }
  if (path.isAbsolute(sourceDir)) {
    return path.join(sourceDir, file);
  }
  if (path.isAbsolute(rootDir)) {
    return path.join(rootDir, sourceDir, file);
  }
  return path.resolve(rootDir, sourceDir, file);
};

export const resolveAbsolutePathForFolder = ({
  folder,
  rootDir,
}: {
  folder: string;
  rootDir: string;
}): string => {
  if (path.isAbsolute(folder)) {
    return folder;
  }
  if (path.isAbsolute(rootDir)) {
    return path.join(rootDir, folder);
  }
  return path.resolve(rootDir, folder);
};
