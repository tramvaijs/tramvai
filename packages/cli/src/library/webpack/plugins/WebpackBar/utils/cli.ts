import chalk from 'chalk';

import textTable from 'text-table';
import { range } from '.';

import { BLOCK_CHAR, BLOCK_CHAR2, BAR_LENGTH } from './consts';

export const colorize = (color) => {
  if (color[0] === '#') {
    return chalk.hex(color);
  }

  return chalk[color] || chalk.keyword(color);
};

export const renderBar = (progress, color) => {
  const w = progress * (BAR_LENGTH / 100);
  const bg = chalk.white(BLOCK_CHAR);
  const fg = colorize(color)(BLOCK_CHAR2);

  return range(BAR_LENGTH)
    .map((i) => (i < w ? fg : bg))
    .join('');
};

export function createTable(data) {
  return textTable(data, {
    align: data[0].map(() => 'l'),
  }).replace(/\n/g, '\n\n');
}

export function ellipsis(str, n) {
  if (str.length <= n - 3) {
    return str;
  }
  return `${str.substr(0, n - 1)}...`;
}

export function ellipsisLeft(str, n) {
  if (str.length <= n - 3) {
    return str;
  }
  return `...${str.substr(str.length - n - 1)}`;
}
