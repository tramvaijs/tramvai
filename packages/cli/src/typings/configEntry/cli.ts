import type { Configuration } from 'webpack';
import type { DeduplicateStrategy } from '@tinkoff/webpack-dedupe-plugin';
import type { ConfigEntry, OverridableOption } from './common';

export type TranspilationLoader = 'babel' | 'swc';
export type Minifier = 'terser' | 'swc';

type SimplifiedSvgoConfig = string | { name: string; params: Record<string, any> };
type Notifications = {
  /**
   * @title Defines when success notifications are shown
   */
  suppressSuccess?: false | true | 'always' | 'initial';
  /**
   * @title True to suppress the warning notifications, otherwise false
   */
  suppressWarning?: boolean;
  /**
   * @title True to activate (focus) the terminal window when a compilation error occurs
   */
  activateTerminalOnError?: boolean;
};

export type WebpackExperiments = Omit<
  Configuration['experiments'],
  'buildHttp' | 'lazyCompilation' | 'css'
> & {
  /**
   * @title Enable additional in-memory caching of modules which are unchanged and reference only unchanged modules.
   * @default true
   */
  cacheUnaffected: boolean;
  /**
   * @title Enable backward compatibility with webpack previous major versions
   * @default false
   */
  backCompat: boolean;
};

export type MinicssExperiments = {
  /**
   * @title Use a new webpack API to execute modules instead of child compilers. This improves performance and memory usage a lot.
   * @default true
   */
  useImportModule: OverridableOption<boolean>;
};

export type TranspilationExperiments = {
  /**
   * @title specify loader to transpile js-ts code
   * @default "babel"
   */
  loader: OverridableOption<TranspilationLoader>;
};

export interface Experiments {
  /**
   * @title experiments configuration for [webpack](https://webpack.js.org/configuration/experiments/)
   * @default {}
   */
  webpack: WebpackExperiments;
  /**
   * @title experimental settings for [mini-css-extract-plugin](https://github.com/webpack-contrib/mini-css-extract-plugin)
   * @default {}
   */
  minicss: MinicssExperiments;

  /**
   * @title experimental settings for code transpilation
   * @default {}
   */
  transpilation: TranspilationExperiments;
  /**
   * @title specify minifer for terser plugin
   * @default "terser"
   */
  minifier: OverridableOption<Minifier>;
}

/**
 * @default {}
 * @additionalProperties true
 */
type Define = Record<string, string>;

/**
 * @tutorial
 * - Common properties that are defined in the tramvai.json config file
 * - These properties are later used to generate the json-schema
 * - Although, some of the marked as required in typings they are anyway converted to optional
 * while generating schema
 * - you can provide default value with "@default" jsdoc in order to replace with default on schema validation
 * - if you provide default value with "@default" mark the property as required to easier its usage in cli code
 * - if the option doesn't have a default value mark it as optional to properly handle this case in cli code
 * - to specify option that might be overridden for dev/prod environments specify its type as `OverridableOption<T>`. You can
 * provide the default value either as single value (will be used in both envs) or in the form of { development: T, production: T } (to be able to
 * provide defaults for all of the envs)
 * - do not use `OverridableOption<T>` with complex types that has internal properties with "@default" as that defaults will not be set by ajv
 */
export interface CliConfigEntry extends ConfigEntry {
  /**
   * @title Enable source maps
   * @default false
   */
  sourceMap: OverridableOption<boolean>;
  /**
   * @title Change different experimental cli settings
   * @default {}
   */
  experiments: Experiments;
  /**
   * @title List of modules to exclude from `@babel/preset-env`
   * @description Option doesn't affect build with swc loader
   */
  excludesPresetEnv?: string[];
  /**
   * @title Webpack's thread-loader config
   * @additionalProperties true
   */
  threadLoader?: Record<string, unknown>;
  /**
   * @title Define static variables that should be replaced by name with provided value
   * @default {}
   */
  define: {
    shared: Define;
    development: Define;
    production: Define;
  };
  /**
   * @title `@deprecated` will be removed in next major release.
   * @default false
   */
  generateDataQaTag: boolean;
  /**
   * @title `@deprecated` will be removed in next major release. Включает использование плагина fill-action-name
   * @default false
   */
  enableFillActionNamePlugin: boolean;
  /**
   * @default {}
   */
  postcss: {
    /**
     * @title Path to postcss config file. By default, `postcss.config.js` file is used
     */
    config?: string;
    /**
     * @title CSS identifiers build algorithm
     */
    cssLocalIdentName?: OverridableOption<string>;
    /**
     * @title Path to postcss config file for assets
     */
    assetsConfig?: string;

    /**
     * @title Enable CSS modules for all files matching /RegExp/i.test(filename) regexp.
     */
    cssModulePattern?: string;
  };
  /**
   * @title `@deprecated` as cli now supports baseUrl and paths from the app's tsconfig.json file.
   * Just check or add configuration to your tsconfig file and remove alias from tramvai.json
   * @additionalProperties true
   */
  alias?: Record<string, any>;
  /**
   * @title svgo-loader options
   */
  svgo?: {
    /**
     * @title svgo plugins, you can find full types in PluginConfig inside `svgo` source code,
     * but this interface too complex and heavy for schema
     */
    plugins?: SimplifiedSvgoConfig[];
  };

  /**
   * @title Settings for image-webpack-loader
   */
  imageOptimization?: {
    /**
     * @title Enable image-webpack-loader
     */
    enabled?: boolean;
    /**
     * @title Pass options to image-webpack-loader
     * @additionalProperties true
     */
    options?: Record<string, any>;
  };
  /**
   * @title transpile libs based only on %40tinkoff/is-modern-lib
   * @default true
   */
  transpileOnlyModernLibs: boolean;
  /**
   * @title Webpack specific settings
   * @default {}
   */
  webpack: {
    /**
     * @title Browser package resolve aliases. E.g. { "stream": "stream-browserify" }
     * @additionalProperties true
     */
    resolveAlias?: Record<string, string>;
    /**
     * @title Browser packages to provide with ProvidePlugin. E.g. { "Buffer": ["buffer", "Buffer"] }
     * @additionalProperties true
     */
    provide?: Record<string, any>;
    /**
     * @title Configure https://webpack.js.org/configuration/watch/#watchoptions
     * @description For OSX users and big projects it's recommended to enable watch polling, e.g. `{ poll: 1000 }`, more info - https://github.com/webpack/watchpack/issues/222
     */
    watchOptions?: Configuration['watchOptions'];
    /**
     * @title Use the specified type of source maps for building in development mode
     */
    devtool: 'eval' | 'eval-cheap-source-map' | 'eval-cheap-module-source-map' | 'eval-source-map';
  };

  // options that affect only production builds
  /**
   * @title Enable DedupePlugin
   * @default {}
   */
  dedupe: {
    /**
     * @title Strategy for DedupePlugin
     * @default true
     */
    enabled: boolean;
    /**
     * @title Does DedupePlugin should be enabled in development mode
     * Why it might be useful see [issue](https://github.com/Tinkoff/tramvai/issues/11)
     * @default false
     */
    enabledDev: boolean;
    /**
     * @title Strategy for DedupePlugin
     * @default "equality"
     */
    strategy: DeduplicateStrategy;
    /**
     * @title Sets ignore to DedupePlugin
     */
    ignore?: string[];
  };
  /**
   * @title terser settings
   * @default {}
   */
  terser: {
    /**
     * @title Configuration for enabling parallel compression for terser plugin
     * @default true
     */
    parallel: boolean;
  };
  /**
   * @title Set minimizer for css
   * @default "css-minimizer"
   */
  cssMinimize: 'csso' | 'css-minimizer';

  // options that affect only development builds

  /**
   * @title React hot-refresh
   * @default {}
   */
  hotRefresh: {
    /**
     * @title Enable react hot-refresh
     * @default true
     */
    enabled?: boolean;
    /**
     * @title Configure react hot-refresh https://github.com/pmmmwh/react-refresh-webpack-plugin#options
     * @default {}
     */
    options?: {
      /**
       * @default false
       */
      overlay?: boolean | Record<string, any>;
      [key: string]: any;
    };
  };
  /**
   * @title Controls build notifications settings during development
   * @default {}
   */
  notifications: Notifications & {
    client?: Notifications;
    server?: Notifications;
  };
  /**
   * @title Specify dependencies that will be shared between application and child-apps
   * @description Properly defining that dependencies may greatly reduce filesize of loaded js on the client
   * @default {}
   */
  shared: {
    /**
     * @title Should default dependencies list be added to shared list
     * @description It includes the list of commonly used dependencies in the child-apps
     * By default, it is enabled in application in case of tramvai/module-child-app is specified in package.json
     * and for child-apps
     */
    defaultTramvaiDependencies?: boolean;
    /**
     * @title add caret range specifier for tramvai dependencies
     * @description minimal versions are inferred from package.json
     * @default true
     */
    flexibleTramvaiVersions: boolean;
    /**
     * @title add chunks to preload as critical in-parallel with "platform.js"
     * @description this option is useful when you need to create async boundary for app dependencies.
     * More info - https://webpack.js.org/concepts/module-federation/#uncaught-error-shared-module-is-not-available-for-eager-consumption
     * @default []
     */
    criticalChunks: string[];
    /**
     * @title list of the dependencies that will be shared
     * @default []
     */
    deps: Array<
      | string
      | {
          /**
           * @title name of the dependency import
           */
          name: string;
          /**
           * @title if dependency is marked as singleton the dependency will be initialized only once and will not be updated
           * @description Do not overuse that feature as it may lead to subtle bugs in case of different versions on different sides
           * @default false
           */
          singleton: boolean;
        }
    >;
  };
}
