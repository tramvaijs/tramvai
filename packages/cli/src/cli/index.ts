import path from 'path';
import { resolvePackageManager, NpmPackageManager } from '@tinkoff/package-manager-wrapper';
import { CLI } from './CLI';
import { Analytics } from '../models/analytics';
import { Logger } from '../models/logger';
import { ConfigManager } from '../models/config';
import { initSentry } from '../utils/sentry';

import type { CommandMap } from '../models/command';

import buildCommand from '../commands/build/command';
import { StartCommand } from '../commands/start/command';
import lintCommand from '../commands/lint/command';
import taskCommand from '../commands/task/command';
import analyze from '../commands/analyze/command';
import generate from '../commands/generate/command';
import newCommand from '../commands/new/command';
import updateCommand from '../commands/update/command';
import addCommand from '../commands/add/command';
import { StaticCommand } from '../commands/static/command';
import { StartProdCommand } from '../commands/start-prod/command';
import { BenchmarkCommand } from '../commands/benchmark/command';

import npmInstallTask from '../tasks/npm-install';
import buildAppTask from '../tasks/build-app';
import type { TaskMap } from '../models/task';
import { getRootFile } from '../utils/getRootFile';
import { getTramvaiConfig } from '../utils/getTramvaiConfig';
import { syncJsonFile } from '../utils/syncJsonFile';

async function loadCommands(): Promise<CommandMap> {
  return [
    buildCommand,
    StartProdCommand,
    StartCommand,
    lintCommand,
    taskCommand,
    analyze,
    generate,
    newCommand,
    updateCommand,
    addCommand,
    StaticCommand,
    BenchmarkCommand,
  ];
}

async function loadTasks(): Promise<TaskMap> {
  return [npmInstallTask, buildAppTask];
}

const defaultPackageInfo = { name: 'init app', version: '0.0.1' };

export async function cliInitialized(cliArgs = process.argv) {
  const logger = new Logger();
  const sentry = initSentry();

  try {
    const commandContext = {
      command: cliArgs[2] || '',
      argv: cliArgs,
    };
    sentry.setContext('command', commandContext);
    const commandsMap = await loadCommands();
    const tasksMap = await loadTasks();

    const { content: packageInfo = defaultPackageInfo } = getRootFile<{
      name: string;
      version: string;
    }>('package.json');
    const { content: config } = getTramvaiConfig();

    const analytic = new Analytics({
      packageInfo,
      trackingCode: 'UA-122261674-1',
    });

    const configManager = new ConfigManager({ config, syncConfigFile: syncJsonFile });
    const packageManager = resolvePackageManager({ rootDir: process.cwd() });

    const cliRootDir = path.resolve(__dirname, '../', '../');
    const cliPackageManager = new NpmPackageManager({
      rootDir: cliRootDir,
    });

    const cliInstance = new CLI(
      commandsMap,
      tasksMap,
      logger,
      configManager,
      analytic,
      cliRootDir,
      cliPackageManager,
      packageManager
    );

    return await cliInstance.run(cliArgs);
  } catch (e: any) {
    logger.event({
      type: 'error',
      event: 'GLOBAL:ERROR',
      message: e.message || e,
      payload: e.details || e.stack,
    });
    sentry.captureException(e);

    throw e;
  } finally {
    await sentry.close();
  }
}
