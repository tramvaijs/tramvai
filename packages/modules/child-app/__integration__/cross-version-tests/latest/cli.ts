import { start } from '@tramvai/cli';
import { startCli } from '@tramvai/test-integration';
import { resolve } from 'path';
import detectPort from 'detect-port';

const EXAMPLE_DIR = resolve(__dirname, '..', '..', '..', '..', '..', '..', 'examples', 'child-app');

export const startRootApp = ({
  define,
  env,
}: {
  define: Record<string, string>;
  env: Record<string, string>;
}) => {
  return startCli(
    {
      name: 'root-app',
      type: 'application',
      root: resolve(EXAMPLE_DIR, 'root-app'),
      shared: {
        deps: ['@tramvai/react-query', '@tramvai/module-react-query'],
      },
      define: {
        development: define,
      },
    },
    {
      rootDir: EXAMPLE_DIR,
      env,
      resolveSymlinks: false,
      fileCache: true,
    }
  );
};

export const startChildApp = async (
  name: string,
  {
    shared = {},
  }: {
    shared?: { deps?: string[] };
  } = {}
) => {
  const port = await detectPort(0);
  const staticPort = await detectPort(0);

  return start({
    config: {
      type: 'child-app',
      root: resolve(EXAMPLE_DIR, 'child-apps', name),
      name,
      hotRefresh: {
        enabled: true,
      },
      shared,
    },
    port,
    staticPort,
    rootDir: EXAMPLE_DIR,
    resolveSymlinks: false,
  });
};
