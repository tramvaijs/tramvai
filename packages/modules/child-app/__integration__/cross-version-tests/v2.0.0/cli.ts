import { start } from '@tramvai/cli';
import { startCli } from '@tramvai/test-integration';
import { resolve } from 'path';
import detectPort from 'detect-port';

const EXAMPLE_DIR = __dirname;

export const startRootApp = ({
  define,
  env,
}: {
  define: Record<string, string>;
  env: Record<string, string>;
}) => {
  return startCli(
    {
      name: 'root-app',
      type: 'application',
      root: resolve(EXAMPLE_DIR, 'root-app'),
      commands: {
        build: {
          options: {
            server: resolve(__dirname, 'root-app'),
          },
          configurations: {
            definePlugin: {
              dev: define,
            },
          },
        },
      },
    },
    {
      rootDir: EXAMPLE_DIR,
      env,
      resolveSymlinks: false,
    }
  );
};

export const startChildApp = async (
  name: string,
  {
    shared = {},
  }: {
    shared?: { deps?: string[] };
  } = {}
) => {
  const port = await detectPort(0);
  const staticPort = await detectPort(0);

  return start({
    config: {
      type: 'child-app',
      root: resolve(EXAMPLE_DIR, 'child-apps', name),
      name,
      commands: {
        serve: {
          configurations: {
            hotRefresh: true,
          },
        },
      },
    },
    port,
    staticPort,
    rootDir: EXAMPLE_DIR,
    resolveSymlinks: false,
  });
};
