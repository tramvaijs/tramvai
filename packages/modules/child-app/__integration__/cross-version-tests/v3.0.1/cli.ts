import { start } from '@tramvai/cli';
import { startCli } from '@tramvai/test-integration';
import { resolve } from 'path';
import detectPort from 'detect-port';

const EXAMPLE_DIR = __dirname;

export const startRootApp = ({
  define,
  env,
}: {
  define: Record<string, string>;
  env: Record<string, string>;
}) => {
  return startCli(
    {
      name: 'root-app',
      type: 'application',
      root: resolve(EXAMPLE_DIR, 'root-app'),
      define: {
        development: define,
      },
    },
    {
      rootDir: EXAMPLE_DIR,
      env,
      resolveSymlinks: false,
      fileCache: true,
    }
  );
};

export const startChildApp = async (
  name: string,
  {
    shared = {},
  }: {
    shared?: { deps?: string[] };
  } = {}
) => {
  const port = await detectPort(0);
  const staticPort = await detectPort(0);

  return start({
    config: {
      type: 'child-app',
      root: resolve(EXAMPLE_DIR, 'child-apps', name),
      name,
      hotRefresh: {
        enabled: true,
      },
    },
    port,
    staticPort,
    rootDir: EXAMPLE_DIR,
    resolveSymlinks: false,
  });
};
