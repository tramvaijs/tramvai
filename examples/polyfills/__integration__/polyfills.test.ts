import path from 'path';
import fs from 'fs';
import { build } from '@tramvai/cli';
import type { StatsCompilation } from 'webpack';

jest.setTimeout(60000);

describe('polyfills', () => {
  it('stable list of polyfills', async () => {
    await build({
      rootDir: path.resolve(__dirname, '../'),
      target: 'polyfills',
      modern: false,
      withModulesStats: true,
      buildType: 'client',
      fileCache: false,
    });

    const distClientDirectory = path.resolve(__dirname, '../dist', 'client');
    const statsFile = fs.readFileSync(
      path.join(distClientDirectory, 'stats-modules.json'),
      'utf-8'
    );
    const stats: StatsCompilation = JSON.parse(statsFile);

    const polyfillChunk = stats.chunks?.find((chunk) => chunk.id === 'polyfill');

    const polyfillChunkFilename = polyfillChunk?.files?.[0];
    expect(polyfillChunkFilename).toBeTruthy();

    const polyfillChunkModules = polyfillChunk?.modules;
    expect(polyfillChunkModules).toBeTruthy();

    const polyfillModules = polyfillChunkModules!
      .map((item) => item.name)
      .filter((name) => name && !/core-js\/internals/.test(name));

    expect({
      polyfills: polyfillModules,
      polyfillsCount: polyfillModules.length,
    }).toMatchSnapshot('default');
  });
});
