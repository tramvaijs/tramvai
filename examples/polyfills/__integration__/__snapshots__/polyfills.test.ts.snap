// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`polyfills stable list of polyfills: default 1`] = `
{
  "polyfills": [
    "../../examples/polyfills/src/polyfill.ts + 11 modules",
    "../../node_modules/core-js/modules/es.aggregate-error.constructor.js",
    "../../node_modules/core-js/modules/es.aggregate-error.js",
    "../../node_modules/core-js/modules/es.array.at.js",
    "../../node_modules/core-js/modules/es.array.concat.js",
    "../../node_modules/core-js/modules/es.array.copy-within.js",
    "../../node_modules/core-js/modules/es.array.fill.js",
    "../../node_modules/core-js/modules/es.array.filter.js",
    "../../node_modules/core-js/modules/es.array.find-index.js",
    "../../node_modules/core-js/modules/es.array.find.js",
    "../../node_modules/core-js/modules/es.array.flat-map.js",
    "../../node_modules/core-js/modules/es.array.flat.js",
    "../../node_modules/core-js/modules/es.array.from.js",
    "../../node_modules/core-js/modules/es.array.includes.js",
    "../../node_modules/core-js/modules/es.array.index-of.js",
    "../../node_modules/core-js/modules/es.array.iterator.js",
    "../../node_modules/core-js/modules/es.array.last-index-of.js",
    "../../node_modules/core-js/modules/es.array.map.js",
    "../../node_modules/core-js/modules/es.array.of.js",
    "../../node_modules/core-js/modules/es.array.reduce-right.js",
    "../../node_modules/core-js/modules/es.array.reduce.js",
    "../../node_modules/core-js/modules/es.array.reverse.js",
    "../../node_modules/core-js/modules/es.array.slice.js",
    "../../node_modules/core-js/modules/es.array.sort.js",
    "../../node_modules/core-js/modules/es.array.species.js",
    "../../node_modules/core-js/modules/es.array.splice.js",
    "../../node_modules/core-js/modules/es.array.unscopables.flat-map.js",
    "../../node_modules/core-js/modules/es.array.unscopables.flat.js",
    "../../node_modules/core-js/modules/es.json.stringify.js",
    "../../node_modules/core-js/modules/es.json.to-string-tag.js",
    "../../node_modules/core-js/modules/es.map.constructor.js",
    "../../node_modules/core-js/modules/es.map.js",
    "../../node_modules/core-js/modules/es.math.to-string-tag.js",
    "../../node_modules/core-js/modules/es.object.assign.js",
    "../../node_modules/core-js/modules/es.object.define-getter.js",
    "../../node_modules/core-js/modules/es.object.define-setter.js",
    "../../node_modules/core-js/modules/es.object.entries.js",
    "../../node_modules/core-js/modules/es.object.freeze.js",
    "../../node_modules/core-js/modules/es.object.from-entries.js",
    "../../node_modules/core-js/modules/es.object.get-own-property-descriptor.js",
    "../../node_modules/core-js/modules/es.object.get-own-property-descriptors.js",
    "../../node_modules/core-js/modules/es.object.get-own-property-names.js",
    "../../node_modules/core-js/modules/es.object.get-own-property-symbols.js",
    "../../node_modules/core-js/modules/es.object.get-prototype-of.js",
    "../../node_modules/core-js/modules/es.object.has-own.js",
    "../../node_modules/core-js/modules/es.object.is-extensible.js",
    "../../node_modules/core-js/modules/es.object.is-frozen.js",
    "../../node_modules/core-js/modules/es.object.is-sealed.js",
    "../../node_modules/core-js/modules/es.object.keys.js",
    "../../node_modules/core-js/modules/es.object.lookup-getter.js",
    "../../node_modules/core-js/modules/es.object.lookup-setter.js",
    "../../node_modules/core-js/modules/es.object.prevent-extensions.js",
    "../../node_modules/core-js/modules/es.object.seal.js",
    "../../node_modules/core-js/modules/es.object.to-string.js",
    "../../node_modules/core-js/modules/es.object.values.js",
    "../../node_modules/core-js/modules/es.promise.all-settled.js",
    "../../node_modules/core-js/modules/es.promise.all.js",
    "../../node_modules/core-js/modules/es.promise.any.js",
    "../../node_modules/core-js/modules/es.promise.catch.js",
    "../../node_modules/core-js/modules/es.promise.constructor.js",
    "../../node_modules/core-js/modules/es.promise.finally.js",
    "../../node_modules/core-js/modules/es.promise.js",
    "../../node_modules/core-js/modules/es.promise.race.js",
    "../../node_modules/core-js/modules/es.promise.reject.js",
    "../../node_modules/core-js/modules/es.promise.resolve.js",
    "../../node_modules/core-js/modules/es.reflect.to-string-tag.js",
    "../../node_modules/core-js/modules/es.regexp.exec.js",
    "../../node_modules/core-js/modules/es.set.constructor.js",
    "../../node_modules/core-js/modules/es.set.js",
    "../../node_modules/core-js/modules/es.string.at-alternative.js",
    "../../node_modules/core-js/modules/es.string.code-point-at.js",
    "../../node_modules/core-js/modules/es.string.ends-with.js",
    "../../node_modules/core-js/modules/es.string.from-code-point.js",
    "../../node_modules/core-js/modules/es.string.includes.js",
    "../../node_modules/core-js/modules/es.string.iterator.js",
    "../../node_modules/core-js/modules/es.string.match-all.js",
    "../../node_modules/core-js/modules/es.string.match.js",
    "../../node_modules/core-js/modules/es.string.pad-end.js",
    "../../node_modules/core-js/modules/es.string.pad-start.js",
    "../../node_modules/core-js/modules/es.string.raw.js",
    "../../node_modules/core-js/modules/es.string.repeat.js",
    "../../node_modules/core-js/modules/es.string.replace-all.js",
    "../../node_modules/core-js/modules/es.string.replace.js",
    "../../node_modules/core-js/modules/es.string.search.js",
    "../../node_modules/core-js/modules/es.string.split.js",
    "../../node_modules/core-js/modules/es.string.starts-with.js",
    "../../node_modules/core-js/modules/es.string.trim-end.js",
    "../../node_modules/core-js/modules/es.string.trim-left.js",
    "../../node_modules/core-js/modules/es.string.trim-right.js",
    "../../node_modules/core-js/modules/es.string.trim-start.js",
    "../../node_modules/core-js/modules/es.string.trim.js",
    "../../node_modules/core-js/modules/es.symbol.async-iterator.js",
    "../../node_modules/core-js/modules/es.symbol.constructor.js",
    "../../node_modules/core-js/modules/es.symbol.description.js",
    "../../node_modules/core-js/modules/es.symbol.for.js",
    "../../node_modules/core-js/modules/es.symbol.has-instance.js",
    "../../node_modules/core-js/modules/es.symbol.is-concat-spreadable.js",
    "../../node_modules/core-js/modules/es.symbol.iterator.js",
    "../../node_modules/core-js/modules/es.symbol.js",
    "../../node_modules/core-js/modules/es.symbol.key-for.js",
    "../../node_modules/core-js/modules/es.symbol.match-all.js",
    "../../node_modules/core-js/modules/es.symbol.match.js",
    "../../node_modules/core-js/modules/es.symbol.replace.js",
    "../../node_modules/core-js/modules/es.symbol.search.js",
    "../../node_modules/core-js/modules/es.symbol.species.js",
    "../../node_modules/core-js/modules/es.symbol.split.js",
    "../../node_modules/core-js/modules/es.symbol.to-primitive.js",
    "../../node_modules/core-js/modules/es.symbol.to-string-tag.js",
    "../../node_modules/core-js/modules/es.symbol.unscopables.js",
    "../../node_modules/core-js/modules/es.weak-map.constructor.js",
    "../../node_modules/core-js/modules/es.weak-map.js",
    "../../node_modules/core-js/modules/es.weak-set.constructor.js",
    "../../node_modules/core-js/modules/es.weak-set.js",
    "../../node_modules/core-js/modules/web.dom-collections.iterator.js",
    "../../node_modules/core-js/modules/web.queue-microtask.js",
    "../../node_modules/core-js/modules/web.url-search-params.constructor.js",
    "../../node_modules/core-js/modules/web.url-search-params.js",
    "../../node_modules/core-js/modules/web.url.constructor.js",
    "../../node_modules/core-js/modules/web.url.js",
    "../../node_modules/core-js/modules/web.url.to-json.js",
    "../../node_modules/resize-observer-polyfill/dist/ResizeObserver.es.js",
    "../libs/main-polyfills/node_modules/intersection-observer/intersection-observer.js",
    "webpack/runtime/global",
  ],
  "polyfillsCount": 123,
}
`;
